attestation-agent/attestation-agent/src/bin/grpc-aa/main.rs

```diff
@@ -70,6 +83,24 @@ pub async fn main() -> Result<()> {
             }
         }
     }
+    if let Some(initdata_toml) = cli.initdata_toml {
+        info!("Initdata TOML file is given by parameter, try to check.");
+        let initdata =
+            std::fs::read_to_string(&initdata_toml).context("read initdata toml file")?;
+        let (_, digest) = attester::Initdata::parse_and_get_digest(&initdata)
+            .context("parse initdata from toml")?;
+        info!("Initdata digest: {:?}", digest);
+        let res = aa.bind_init_data(&digest).await.context(
+            "The initdata supplied by the parameter is inconsistent with that of the current platform.",
+        )?;
+        match res {
+            attester::InitDataResult::Ok => info!("Check initdata passed."),
+            attester::InitDataResult::Unsupported => {
+                info!("Platform does not support initdata checking. Jumping.")
+            }
+        }
+        aa.set_initdata(initdata);
+    }
```

Comment:
Should this code block be wrapped into a generic method? As I notice other place also use the same process handling logic.